### PR TITLE
Hw 5, arch docs and phpat tests + some minor fixes

### PR DESCRIPTION
**Pull request by morchis27**

### CODE REVIEW

File: tests/Architecture/ArchitectureTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public function testModelDoesNotDependOnController(): Rule
+    {
+        return PHPat::rule()
+            ->classes(Selector::inNamespace('App\Models'))
+            ->shouldNotDependOn()
+            ->classes(
+                Selector::inNamespace('App\Controller'),
+            )
+            ->because('service must not depend on any class that is utilizing service to perform some task');
```
Comment (rgxhomie) [inline tests/Architecture/ArchitectureTest.php]: This does not seem to be checking services, but model-controller relations. Just a typo, I suppose

File: docs/architecture.png
Comment (rgxhomie) [inline docs/architecture.png:1]: Neat 🥇

**Review by rgxhomie — APPROVED** [review]
Looks really well!

File: tests/Architecture/ArchitectureTest.php (hunk 1 above)
Comment (morchis27) [reply tests/Architecture/ArchitectureTest.php]: fixed

**Review by Artyomushko — APPROVED** [review]
Дуже гарна робота

**Pull request merged**